kajas/src/app/login/login.component.ts

```diff
@@ -0,0 +1,123 @@
+import { Component, OnInit, OnDestroy, Renderer2 } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { HttpClient } from '@angular/common/http';
+import { Router } from '@angular/router';
+import { SessionStorageService } from 'angular-web-storage';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css'],
+})
+export class LoginComponent implements OnInit, OnDestroy {
+  loginForm: FormGroup;
+  hidePassword: boolean = true;
+  showModal: boolean = false;
+  modalMessage: string = '';
+
+  constructor(
+    private renderer: Renderer2,
+    private fb: FormBuilder,
+    private http: HttpClient,
+    private router: Router,
+    private sessionStorage: SessionStorageService
+  ) {
+    this.loginForm = this.fb.group({
+      email: ['', [Validators.required, Validators.email]],
+      password: ['', [Validators.required, Validators.minLength(6)]],
+    });
+  }
+
+  ngOnInit(): void {
+    this.setInitialStyles();
+    window.addEventListener('resize', this.applyBackground.bind(this));
```

Comment:
### Stop using Vanilla JS way of doing things.
- Get rid of using querySelectors and addEventsListener.
[https://docs.google.com/presentation/d/1Mdcq3WcHksBlbCvxi6wx5vajn4v8LGlJRnBIu3x4fvw/edit#slide=id.g2e205e48566_0_29](url)